### PR TITLE
Issue#9948 refactor TablesContext

### LINKED ISSUE #9948: refactor TablesContext
Using stream to  **Simplify** the code and **keep it consistent** with other functions in TablesContext.

### PR DESCRIPTION
**Pull request by MingxingLAI**
Fixes #9948 .

Changes proposed in this pull request:

1. getTableNames function
2. TablesContext constructor

### CODE REVIEW

File: shardingsphere-infra/shardingsphere-infra-binder/src/main/java/org/apache/shardingsphere/infra/binder/segment/table/TablesContext.java [hunk 1]
```diff
@@ -46,14 +46,9 @@ public TablesContext(final SimpleTableSegment tableSegment) {
...
-        Map<String, SimpleTableSegment> tableMaps = new HashMap<>(1, 1);
-        Collection<SimpleTableSegment> actualTables = new LinkedList<>();
-        for (SimpleTableSegment each : tableSegments) {
-            if (!tableMaps.containsKey(each.getTableName().getIdentifier().getValue())) {
-                tableMaps.put(each.getTableName().getIdentifier().getValue(), each);
-                actualTables.add(each);
-            }
-        }
+        Collection<SimpleTableSegment> actualTables = new LinkedList<>(tableSegments);
+        Set<String> seen = new HashSet<>();
```
Comment (terrymanu) [inline shardingsphere-infra/shardingsphere-infra-binder/src/main/java/org/apache/shardingsphere/infra/binder/segment/table/TablesContext.java:50]: What is `seen` mean?

**Review by terrymanu — CHANGES_REQUESTED** [review]

**Comment by tristaZero** [comment]
Hi any progress?

**Comment by MingxingLAI** [comment]
I have explained the modification of the code, I don’t know what to do next, and I need help from you. @tristaZero

File: shardingsphere-infra/shardingsphere-infra-binder/src/main/java/org/apache/shardingsphere/infra/binder/segment/table/TablesContext.java (hunk 1 above)
Comment (tristaZero) [reply shardingsphere-infra/shardingsphere-infra-binder/src/main/java/org/apache/shardingsphere/infra/binder/segment/table/TablesContext.java:50]: Have you clarified for this question? @MingxingLAI

**Review by tristaZero — CHANGES_REQUESTED** [review]

**Comment by MingxingLAI** [comment]
I think it has been explained, and ran all the unit tests, you can help me look at the code.

**Comment by tristaZero** [comment]
@terrymanu Please review this one. It is left to stand for a long time.

**Comment by MingxingLAI** [comment]
Any suggestion or question ? @tristaZero @terrymanu

**Comment by terrymanu** [comment]
> Any suggestion or question ? @tristaZero @terrymanu

Hi, What is variable name `seen` mean?

**Comment by MingxingLAI** [comment]
> > Any suggestion or question ? @tristaZero @terrymanu
> 
> Hi, What is variable name `seen` mean?


Did you notice my previous explanation ？ 

The `seen` set is used for deduplication, and its function is the same as the original HashMap `tableMaps`

- removeIf will remove an element if it meets the specified criteria
- Set.add will return false if it did not modify the Set, i.e. already contains the value

combining these two, it will remove all elements which has been encountered before.

**Comment by tristaZero** [comment]
Hey @MingxingLAI 
This PR looks plain, just a function name for our consideration. Since we do not prefer an acronym name, could you change it? Is it not knotty, right?

**Pull request closed without merging**